packages-exp/auth-exp/src/api/index.test.ts

```diff
@@ -0,0 +1,159 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FirebaseError } from '@firebase/util';
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import { SinonStub, stub, useFakeTimers } from 'sinon';
+import { DEFAULT_API_TIMEOUT, Endpoint, HttpMethod, performApiRequest } from '.';
+import { mockEndpoint } from '../../test/api/helper';
+import { mockAuth } from '../../test/mock_auth';
+import * as mockFetch from '../../test/mock_fetch';
+import { ServerError } from './errors';
+import { AuthErrorCode } from '../core/errors';
+
+use(chaiAsPromised);
+
+describe('performApiRequest', () => {
+  const request = {
+    requestKey: 'request-value'
+  };
+
+  const serverResponse = {
+    responseKey: 'response-value'
+  };
+
+  context('with regular requests', () => {
```

Comment:
Is context an alias to describe?

Reply:
yes, it's just an alias that mocha inherit from rspec https://lmws.net/describe-vs-context-in-rspec

TLDR describe encompasses the functionality you want to test, context groups similar setup blocks for the same functionality

Reply:
Nice!